### PR TITLE
STY: autofix `unsorted-dunder-all` (`RUF022`)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
ref #17430

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
For long `__all__` this is indisputably better, while for short ones it does not matter, so I think we should go with it. Will not immediately merge this one, given how many files are touched.

File: astropy/cosmology/__init__.py
Comment (pllim) [inline astropy/cosmology/__init__.py:1]: @nstarman might have opinions on this one. Looks like he ordered it by category.

Comment (nstarman) [reply astropy/cosmology/__init__.py:1]: Yes! I'd actually like this one to be skipped by ruff.

Comment (neutrinoceros) [reply astropy/cosmology/__init__.py:1]: would you like it ignored at the package level or just cosmology ?

**Comment by nstarman** [comment]
In other projects I've set up sphinx to use __all__ to organize the API docs. I don't think we do that, so then this PR is good. Selection modules can apply ignores as needed.

File: astropy/cosmology/__init__.py
Comment (nstarman) [reply astropy/cosmology/__init__.py:1]: Just cosmology.

Comment (neutrinoceros) [reply astropy/cosmology/__init__.py:1]: done !

File: astropy/coordinates/builtin_frames/__init__.py [hunk 1]
```diff
@@ -78,37 +78,37 @@
 # get included.  Note that the order here determines the order in the
 # documentation of the built-in frames (see make_transform_graphs_docs).
```
Comment (eerovaher) [inline astropy/coordinates/builtin_frames/__init__.py:79]: The comment here is true, but I don't see why the order should matter. @mhvk, why was it important?

Comment (neutrinoceros) [reply astropy/coordinates/builtin_frames/__init__.py:79]: good point. Should I exclude the whole subpacakge, or just this file ?

Comment (eerovaher) [reply astropy/coordinates/builtin_frames/__init__.py:79]: That sounds like an oversight and I think it should be reported to Ruff.

Comment (mhvk) [reply astropy/coordinates/builtin_frames/__init__.py:79]: For now, my tendency would be to just mark this `__all__` with the appropriate `noqa` - that then is also a reminder to fix it at some point.

Comment (neutrinoceros) [reply astropy/coordinates/builtin_frames/__init__.py:79]: adding a noqa here was surprisingly difficult, but it's done !

**Review by taldcroft — APPROVED** [review]
Fine for table, time, io.ascii, misc/yaml.

**Comment by WilliamJamieson** [comment]
If this influences the ordering for the docs, I'm not sure it should be used in modeling.core. `Model` should be the very first thing listed in the docs.

**Comment by neutrinoceros** [comment]
@WilliamJamieson should I revert everything from `modeling` or just `modelling/core.py` ?

File: pyproject.toml [hunk 2]
```diff
@@ -391,6 +391,7 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
 [tool.ruff.lint.extend-per-file-ignores]
 "setup.py" = ["INP001"]  # Part of configuration, not a package.
 ".github/workflows/*.py" = ["INP001"]
+"astropy/cosmology/**/*.py" = ["RUF022"]
```
Comment (nstarman) [inline pyproject.toml:394]: Apologies, I had this marked as viewed. I think we discussed just ignoring the files with manually sorted `__all__`, not the whole module.

Comment (neutrinoceros) [reply pyproject.toml:394]: No worries. Am I right to assume that "`__all__`s with comments from gh-14953" is the exact same set ? I'll push with this interpretation, but please do correct me if needed !

**Review by matteobachetti — APPROVED** [review]
Changes to `io.misc` are trivial

File: pyproject.toml (hunk 2 above)
Comment (nstarman) [reply pyproject.toml]: I think just the top and flrw inits

Comment (nstarman) [reply pyproject.toml]: Thanks!

**Comment by nstarman** [comment]
Is this ready to merge?

**Comment by neutrinoceros** [comment]
AFAIC yes.

**Comment by mhvk** [comment]
Well, I'll push the merge button, then. Thanks, @neutrinoceros!

**Pull request merged**